qiskit/pulse/builder.py

```diff
@@ -0,0 +1,1996 @@
+        default_transpiler_settings=default_transpiler_settings,
+        default_circuit_scheduler_settings=default_circuit_scheduler_settings)
+
+
+# Builder Utilities ############################################################
+class NoActiveBuilder(exceptions.PulseError):
+    """Raised if no builder context is active."""
+
+
+def _active_builder() -> _PulseBuilder:
+    """Get the active builder in the active context.
+
+    Returns:
+        The active active builder in this context.
+
+    Raises:
+        NoActiveBuilder: If a pulse builder function is called outside of a
+            builder context.
+    """
+    try:
+        return BUILDER_CONTEXTVAR.get()
+    except LookupError:
+        raise NoActiveBuilder(
+            'A Pulse builder function was called outside of '
+            'a builder context. Try calling within a builder '
+            'context, eg., "with pulse.build() as schedule: ...".')
+
+
+def active_backend():
+    """Get the backend of the currently active builder context.
+
+    Returns:
+        BaseBackend: The active backend in the currently active
+            builder context.
+
+    Raises:
+        exceptions.BackendNotSet: If the builder does not have a backend set.
+    """
+    builder = _active_builder().backend
+    if builder is None:
+        raise BackendNotSet(
+            'This function requires the active builder to '
+            'have a "backend" set.')
+    return builder
+
+
+def append_block(block: Schedule):
+    """Call a block by appending to the active builder block."""
+    _active_builder().append_block(block)
+
+
+def append_instruction(instruction: instructions.Instruction):
+    """Append an instruction to the current active builder context block.
+
+    Example Usage:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+
+        d0 = pulse.DriveChannel(0)
+
+        with pulse.build() as pulse_prog:
+            pulse.append_instruction(pulse.Delay(10, d0))
+
+        print(pulse_prog.instructions)
+    """
+    _active_builder().append_instruction(instruction)
+
+
+def num_qubits() -> int:
+    """Return number of qubits in the currently active backend.
+
+    Example Usage:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+        from qiskit.test.mock import FakeOpenPulse2Q
+
+        backend = FakeOpenPulse2Q()
+
+        with pulse.build(backend):
+            print(pulse.num_qubits())
+
+    .. note:: Requires the active builder context to have a backend set.
+    """
+    return active_backend().configuration().n_qubits
+
+
+def time_to_samples(time: float) -> int:
+    """Obtain the number of samples that will elapse in ``time`` for the
+    active backend.
+
+    Rounds down.
+
+    Args:
+        time: Time in seconds.
+
+    Returns:
+        The number of samples for the time elapse
+    """
+    return int(time / active_backend().configuration().dt)
+
+
+def samples_to_time(samples: float) -> float:
```

Comment:
Maybe `samples_to_seconds`? Time is a bit vague

Reply:
Sounds good, plus it has alliterations 😄

Reply:
Fixed in 58e7acadb.